odl/tomo/data/mrc2014.py

```diff
@@ -0,0 +1,846 @@
+    axis_order = kwargs.pop('axis_order', (0, 1, 2))
+    if tuple(axis_order) not in permutations((0, 1, 2)):
+        raise ValueError('`axis_order` must be a permutation of (0, 1, 2), '
+                         'got {}'.format(axis_order))
+    dmin = kwargs.pop('dmin', 1.0)
+    dmax = kwargs.pop('dmax', 0.0)
+    dmean = kwargs.pop('dmean', min(dmin, dmax) - 1.0)
+    rms = kwargs.pop('rms', -1.0)
+    mrc_version = kwargs.pop('mrc_version', (2014, 0))
+    if len(mrc_version) != 2:
+        raise ValueError('`mrc_version` must be a sequence of length 2, got '
+                         '{}'.format(mrc_version))
+
+    # Text labels: fill each label up with whitespace to 80 characters.
+    # Create the remaining labels as 80 * '\x00'
+    text_labels_in = kwargs.pop('text_labels', [])
+    nlabl = len(text_labels_in)
+    if nlabl > 10:
+        raise ValueError('expexted maximum of 10 labels, got {} labels'
+                         ''.format(nlabl))
+    text_labels = [str(label).ljust(80) for label in text_labels_in]
+    if any(len(label) > 80 for label in text_labels):
+        raise ValueError('labels cannot have more than 80 characters each')
+
+    # Convert to header-friendly form. Names are required to match
+    # exactly the header field names, and all of them must exist,
+    # so that `eval` below succeeds for all fields.
+    nx, ny, nz = [np.array(n, dtype='int32').reshape([1]) for n in num_cells]
+    mode = np.array(NPY_DTYPE_TO_MRC_MODE[np.dtype(dtype)],
+                    dtype='int32').reshape([1])
+    mx, my, mz = nx, ny, nz
+    cella = (np.array(extent).reshape([3])).astype('float32')
+    mapc, mapr, maps = [np.array(m, dtype='int32').reshape([1]) + 1
+                        for m in axis_order]
+    dmin, dmax, dmean, rms = [np.array(x, dtype='float32').reshape([1])
+                              for x in (dmin, dmax, dmean, rms)]
+    ispg = 1 if kind == 'volume' else 0
+    ispg = np.array(ispg, dtype='int32', ndmin=1)
+    nsymbt = np.array([0], dtype='int32')
+    exttype = np.fromiter('    ', dtype='S1')
+    nversion = np.array(10 * mrc_version[0] + mrc_version[1],
+                        dtype='int32').reshape([1])
+    origin = np.zeros(3, dtype='int32')
+    map = np.fromiter('MAP ', dtype='S1')
+    # TODO: no idea how to properly choose the machine stamp
+    machst = np.fromiter(b'DD  ', dtype='S1')
+    nlabl = np.array(nlabl, dtype='int32').reshape([1])
+    label = np.zeros((10, 80), dtype='S1')  # ensure correct size
+    for i, label_i in enumerate(text_labels):
+        label[i] = np.fromiter(label_i, dtype='S1')
+
+    # Make the header
+    # We use again the specification to set the values
+    header_fields = header_fields_from_table(
+        MRC_2014_SPEC_TABLE, MRC_SPEC_KEYS, MRC_DTYPE_TO_NPY_DTYPE)
+
+    header = OrderedDict()
+    for field in header_fields:
+        header[field['name']] = {'offset': field['offset'],
+                                 'value': eval(field['name'])}
```

Comment:
Overlooked this -- it does give warnings about unused stuff, but saves 20-ish lines of code of the annoying type
```python
header['bla'] = {'offset': header_fields['bla']['offset'], 'value': bla}
```
I could have done it with an explicit dictionary `{'bla': bla}` for all fields. That's fine with me if this looks too "dirty".